### PR TITLE
Create and read entries

### PR DESCRIPTION
**Pull request by karesti**
Includes 3 issues:
- [x] Relooking cache detail. Issue can be closed. https://issues.redhat.com/browse/ISPN-11495 
- [x] Add entry. Don't close yet, follow up PR will come. https://issues.redhat.com/browse/ISPN-11496
- [x] Get using key. Don't close yet. Follow up PR will come. https://issues.redhat.com/browse/ISPN-11497

Things to note:
* The add entry feature does not fully work because a server bug needs to be fixed
* I need to check the KeyContentType and Value Content Type with @gustavonalle 
* The get entry functionality is partially done. The extended information of the entry (expiration and other stuff) needs to be reworked on the server

Does not cover and needs to be done in several other PR and issues
- Recent activity table
- Clear all
- Edit entry
- Delete entry

### CODE REVIEW

**Comment by karesti** [comment]
This issue will evolve with query and other stuff 
For now we only need this base

File: src/services/utils.ts [hunk 1]
```diff
@@ -3,6 +3,51 @@
...
+  CACHE_MODE_LOCAL = 'CACHE_MODE_LOCAL',
+  COMMAND_RETRY = 'COMMAND_RETRY',
+  FAIL_SILENTLY = 'FAIL_SILENTLY',
+  FORCE_ASYNCHRONOUS = 'FORCE_ASYNCHRONOUS',
+  FORCE_SYNCHRONOUS = 'FORCE_SYNCHRONOUS',
+  FORCE_WRITE_LOCK = 'FORCE_WRITE_LOCK',
+  IGNORE_RETURN_VALUES = 'IGNORE_RETURN_VALUES',
+  IGNORE_TRANSACTION = 'IGNORE_TRANSACTION',
+  PUT_FOR_EXTERNAL_READ = 'PUT_FOR_EXTERNAL_READ',
+  PUT_FOR_STATE_TRANSFER = 'PUT_FOR_STATE_TRANSFER',
```
Comment (pruivo) [inline src/services/utils.ts:32]: some flags are internal only and, IMO, should be removed.
`PUT_FOR_STATE_TRANSFER`
`PUT_FOR_X_SITE_STATE_TRANSFER`
`ROLLING_UPGRADE`
`COMMAND_RETRY`

File: data/create-data.sh [hunk 2]
```diff
@@ -1,61 +1,62 @@
...
-curl -XDELETE  http://localhost:11222/rest/v2/caches/programmers
-curl -XDELETE  http://localhost:11222/rest/v2/caches/seniors
-curl -XDELETE  http://localhost:11222/rest/v2/caches/people
-curl -XDELETE  http://localhost:11222/rest/v2/caches/example
-curl -XDELETE  http://localhost:11222/rest/v2/caches/local_users
-curl -XDELETE  http://localhost:11222/rest/v2/caches/invalidated
-curl -XDELETE  http://localhost:11222/rest/v2/caches/invalidated-sync
-curl -XDELETE  http://localhost:11222/rest/v2/caches/xsiteCache
-curl -XDELETE  http://localhost:11222/rest/v2/caches/xsite-transactional
+curl -XDELETE  -u igor:pass http://localhost:11222/rest/v2/caches/users
```
Comment (pruivo) [inline data/create-data.sh:3]: set the username and password in variables so they can be easily changed.

File: data/xsiteCache-transactional.xml [hunk 3]
```diff
@@ -1,11 +1,11 @@
 <infinispan>
   <cache-container>
     <distributed-cache name="xsite-transactional" statistics="true">
-      <backups>
-        <backup site="LON" strategy="SYNC" failure-policy="WARN" timeout="12003">
-          <state-transfer chunk-size="098" timeout="7654" max-retries="321" wait-time="0101"/>
-        </backup>
-      </backups>
+<!--      <backups>-->
```
Comment (pruivo) [inline data/xsiteCache-transactional.xml:4]: are you sure you want to commit this?

**Comment by pruivo** [comment]
Issues found:

### Add Entry:

* Unable to add entries:
Access to fetch at 'http://localhost:11222/rest/v2/caches/xsiteCache/b' from origin 'http://localhost:9000' has been blocked by CORS policy: Request header field key-content-type is not allowed by Access-Control-Allow-Headers in preflight response.

* Max Idle & Time to Live
It says I can set a negative value but it doesn't allow me. It says `-1` isn't a number.

* Key type & Value type
After setting this value, it cannot be unset (even clicking the `X` after the type)

### Get Entry:
If the key isn't set, it returns the metrics. It should be a "nop"

### Random error in console
sometimes, this pops up in the JS console.
```
cacheService.ts?6bac:264 Uncaught (in promise) TypeError: err.text is not a function
    at eval (cacheService.ts?6bac:264)
```

File: src/services/utils.ts (hunk 1 above)
Comment (karesti) [reply src/services/utils.ts:32]: ok

File: data/create-data.sh (hunk 2 above)
Comment (karesti) [reply data/create-data.sh]: done

File: data/xsiteCache-transactional.xml (hunk 3 above)
Comment (karesti) [reply data/xsiteCache-transactional.xml:4]: while the REST server is kaput yes

**Comment by karesti** [comment]
@pruivo I reported all the changes!

Yes the Unable to add entries:
Access to fetch at 'http://localhost:11222/rest/v2/caches/xsiteCache/b' from origin 'http://localhost:9000' has been blocked by CORS policy: Request header field key-content-type is not allowed by Access-Control-Allow-Headers in preflight response.

Is related to 
https://issues.redhat.com/browse/ISPN-11765 
bug in the server that needs to be fixed to work well.
I'm reviewing as well the media type and headers with gustavo.
this will be handled in another PR so that's why I don't want to close get/add issues (but only the first one)

**Comment by pruivo** [comment]
integrated! thanks @karesti

**Pull request merged**